numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -653,6 +653,38 @@ static PyObject *
 /**end repeat**/
 
 static PyObject *
+voidtype_repr(PyObject *self){
+    PyVoidScalarObject *s = (PyVoidScalarObject*) self;
+    PyObject *bytes, *repr;
+    if (PyDataType_HASFIELDS(s->descr)) {
+        return gentype_repr(self);
+    }
+    bytes = PyBytes_FromStringAndSize(s->obval, s->descr->elsize);
+    if (bytes == NULL) {
+        return NULL;
+    }
+    repr = PyObject_Repr(bytes);
+    Py_DECREF(bytes);
+    return repr;
+}
+
+static PyObject *
+voidtype_str(PyObject *self){
+    PyVoidScalarObject *s = (PyVoidScalarObject*) self;
+    PyObject *bytes, *repr;
+    if (PyDataType_HASFIELDS(s->descr)) {
+        return gentype_str(self);
+    }
+    bytes = PyBytes_FromStringAndSize(s->obval, s->descr->elsize);
+    if (bytes == NULL) {
+        return NULL;
+    }
+    repr = PyObject_Repr(bytes);
```

Comment:
This isn't enough - I think that this should be our output:

```
>>> np.array("Hello world").view(np.void)
array(b'\x68\x65\x6C\x6C\x6F\x20\x77\x6F\x72\x6C\x64', dtype=void)
```

Showing ascii characters as text isn't useful - we already have the `bytes_` dtype for that